app/components/IconPicker/components/CustomEmojiPanel.tsx

```diff
@@ -0,0 +1,200 @@
+          setSearchData([
+            {
+              category: DisplayCategory.Search,
+              icons: data?.map((icon) => ({
+                type: IconType.Custom,
+                id: icon.name,
+                value: icon.value,
+              })),
+            },
+          ]);
+          return;
+        }
+      });
+    }
+
+    setSearchData([]);
+  }, [query]);
+
+  const handleEmojiSelection = React.useCallback(
+    // eslint-disable-next-line
+    ({ id, value }: { id: string; value: string }) => {
+      onEmojiChange(value);
+      incrementEmojiCount(id + "-" + value);
+    },
+    [onEmojiChange, incrementEmojiCount]
+  );
+
+  const templateData: DataNode[] = React.useMemo(() => {
+    const freqEmoji = getFrequentEmojis(emojisFreq, lastEmoji);
+
+    return [
+      {
+        category: DisplayCategory.Frequent,
+        icons: freqEmoji.map(({ id, value }) => ({
+          type: IconType.Custom,
+          id,
+          value,
+        })),
+      },
+      {
+        category: DisplayCategory.All,
+        icons: emojis.orderedData.map((emoji) => ({
+          type: IconType.Custom,
+          id: emoji.name,
+          value: emoji.url,
+        })),
+      },
+    ];
+  }, [emojis.orderedData, emojisFreq]);
+
+  React.useLayoutEffect(() => {
+    if (!panelActive) {
+      return;
+    }
+    scrollableRef.current?.scroll({ top: 0 });
+    requestAnimationFrame(() => searchRef.current?.focus());
+  }, [panelActive]);
+
+  return (
+    <Flex column>
+      <UserInputContainer align="center" gap={12}>
+        <StyledInputSearch
+          ref={searchRef}
+          value={query}
+          placeholder={`${t("Search")}…`}
+          onChange={handleFilter}
+        />
+      </UserInputContainer>
+      <GridTemplate
+        ref={scrollableRef}
+        width={panelWidth}
+        height={height - 48}
+        data={searchData.length ? searchData : templateData}
+        onIconSelect={handleEmojiSelection}
+      />
+    </Flex>
+  );
+};
+
+const getFrequentEmojis = (
+  emojisFreq: Record<string, number>,
+  lastEmoji?: string
+) => {
+  const frequent = Object.entries(emojisFreq);
+
+  const emojis = frequent
+    .sort((a, b) => b[1] - a[1])
+    .slice(0, FREQUENTLY_USED_COUNT.Track)
+    .map(([nameUrl]) => {
+      const index = nameUrl.indexOf("-");
+      const id = nameUrl.substring(0, index);
+      const value = nameUrl.substring(index + 1);
+
+      return {
+        id,
+        value,
+      };
+    });
+
+  const isLastPresent = emojis.some(
+    (emoji) => lastEmoji && `${emoji.id}-${emoji.value}` === lastEmoji
+  );
+
+  if (lastEmoji && !isLastPresent) {
+    const index = lastEmoji.indexOf("-");
+    const id = lastEmoji.substring(0, index);
+    const value = lastEmoji.substring(index + 1);
+
+    emojis.pop();
+    emojis.push({
+      id,
+      value,
+    });
+  }
+
+  return emojis.filter((emoji) => isInternalUrl(emoji.value));
+};
+
+const StyledInputSearch = styled(InputSearch)`
```

Comment:
Similar, can we share these instead of repeating? just create `/Components.tsx` file locally is fine